qiskit/aqua/algorithms/phase_estimators/hamiltonian_pe.py

```diff
@@ -0,0 +1,174 @@
+
+    This works for 1Q operators. This is only a stop gap while
+    waiting for a bug fix in Terra. This class does not do Trotterization, etc.
+    It just computes the exact gate.
+    Hopefully, this can be removed soon.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def convert(self, evolved_operator: OperatorBase) -> QuantumCircuit:
+        """Return a circuit for EvolvedOp
+
+        Hmm, looks like the sign in the exponent may be wrong.
+        """
+        from qiskit.quantum_info.synthesis.two_qubit_decompose import two_qubit_cnot_decompose
+        matrix = evolved_operator.to_matrix()
+        if evolved_operator.num_qubits == 1:
+            return self._matrix_to_circuit_1q(matrix)
+        elif evolved_operator.num_qubits == 2:
+            return two_qubit_cnot_decompose(matrix)
+        else:
+            raise ValueError("wrong num qubits")
+
+    def _matrix_to_circuit_1q(self, matrix: numpy.ndarray) -> QuantumCircuit:
+        theta, phi, lam, global_phase = _DECOMPOSER1Q.angles_and_phase(matrix)
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(1)
+        qc._append(U3Gate(theta, phi, lam), [q[0]], [])
+        qc.global_phase = global_phase
+        return qc
+
+
+class HamiltonianPE(PhaseEstimator):
+    r"""Run the Quantum Phase Estimation algorithm to find the eigenvalues of a Hermitian operator.
+
+    This class is nearly the same as :class:`~qiskit.aqua.algorithms.PhaseEstimator`, differing only
+    in that the input in that class is a unitary operator, whereas here the input is a Hermitian
+    operator from which a unitary will be obtained by scaling and exponentiating. The scaling is
+    performed in order to prevent the phases from wrapping around :math:`2\pi`. This class uses and
+    works together with :class:`~qiskit.aqua.algorithms.PhaseEstimationScale` to manage scaling the
+    Hamiltonian and the phases that are obtained by the QPE algorithm. This includes setting, or
+    computing, a bound on the eigenvalues of the operator, using this bound to obtain a scale
+    factor, scaling the operator, and shifting and scaling the measured phases to recover the
+    eigenvalues.
+
+    Note that, although we speak of "evolving" the state according the the Hamiltonian, in the
+    present algorithm, we are not actually considering time evolution. Rather, the role of time is
+    played by the scaling factor, which is chosen to best extract the eigenvalues of the
+    Hamiltonian.
+
+    A few of the ideas in the algorithm may be found in Ref. [1].
+
+    **Reference:**
+
+    [1]: Quantum phase estimation of multiple eigenvalues for small-scale (noisy) experiments
+         T.E. O'Brien, B. Tarasinski, B.M. Terhal
+         https://arxiv.org/abs/1809.09697
+    """
+    def __init__(self,
+                 num_evaluation_qubits: int,
+                 hamiltonian: OperatorBase,
+                 evolution: Optional[Union[EvolutionBase, TempPauliEvolve]] = None,
```

Comment:
Is Optional applicable for `evolution`? In _get_unitary it blindly dereferences it which will fail if it has the default None